nexus/transport.go

```diff
@@ -0,0 +1,675 @@
+	//
+	// Should only be called through an [OperationHandle].
+	//
+	// NOTE: Experimental
+	GetOperationResult(ctx context.Context, options TransportGetOperationResultOptions) (*TransportGetOperationResultResponse[*LazyValue], error)
+
+	// CancelOperation requests to cancel an asynchronous operation.
+	//
+	// Cancelation is asynchronous and may be not be respected by the operation's implementation.
+	//
+	// Should only be called through an [OperationHandle].
+	//
+	// NOTE: Experimental
+	CancelOperation(ctx context.Context, options TransportCancelOperationOptions) (*TransportCancelOperationResponse, error)
+
+	// Close this Transport and release any underlying resources.
+	//
+	// NOTE: Experimental
+	Close() error
+}
+
+// TransportStartOperationResponse is the response to Transport.StartOperation calls. One and only one of Complete or
+// Pending will be populated.
+//
+// NOTE: Experimental
+type TransportStartOperationResponse[T any] struct {
+	// Set when start completes synchronously.
+	//
+	// If T is a [LazyValue], ensure that your consume it or read the underlying content in its entirety and close
+	// it to free up the underlying connection.
+	Complete *OperationResult[T]
+	// Set when the handler indicates that it started an asynchronous operation.
+	// The attached handle can be used to perform actions such as cancel the operation or get its result.
+	Pending *OperationHandle[T]
+	// Links contain information about the operations done by the handler.
+	Links []Link
+}
+
+// TransportGetOperationInfoResponse is the response to Transport.GetOperationInfo calls.
+//
+// NOTE: Experimental
+type TransportGetOperationInfoResponse struct {
+	Info *OperationInfo
+}
+
+// TransportGetOperationResultResponse is the response to Transport.GetOperationResult calls.
+// Use TransportGetOperationResultResponse.GetResult to retrieve the final value or error returned by the operation.
+//
+// NOTE: Experimental
+type TransportGetOperationResultResponse[T any] struct {
+	result *OperationResult[T]
+	Links  []Link
+}
+
+// GetResult returns the final result or error returned by the operation.
+//
+// NOTE: Experimental
+func (gr *TransportGetOperationResultResponse[T]) GetResult() (T, error) {
+	return gr.result.Get()
+}
+
+// TransportCancelOperationResponse is the response to Transport.CancelOperation calls.
+//
+// NOTE: Experimental
+type TransportCancelOperationResponse struct{}
+
+// OperationResult contains the final value or error returned by an operation handler. One and only one of
+// result or err will be populated. Use OperationResult.Get to retrieve the result.
+//
+// In most cases err will be an [OperationError]. Other failures, such as [HandlerError], will be returned by
+// the Transport method called to indicate a failure to communicate with the operation handler.
+//
+// NOTE: Experimental
+type OperationResult[T any] struct {
```

Comment:
Does this need to be exported?

Reply:
I think it's fine since it's it'll move into a separate `transport` package. Just might be a bit awkward to embed a transport type directly.

I would consider exposing the error and the result separately on the transport and exporting this in the main package. Not blocking the PR.